pkg/riff/commands/doctor.go

```diff
@@ -86,21 +142,67 @@ func (*DoctorOptions) checkNamespaces(c *cli.Config) (bool, error) {
 	return ok, nil
 }
 
-func NewDoctorCommand(ctx context.Context, c *cli.Config) *cobra.Command {
-	opts := &DoctorOptions{}
+func checkCustomResourceExistence(c *cli.Config, checks []AccessChecks) (*CrdSummary, error) {
+	var aggregatedStatuses []CrdStatus
+	crds := c.ApiExtensions().CustomResourceDefinitions()
+	for _, check := range checks {
+		serverResource := check.Resource
+		if !serverResource.Custom {
+			continue
+		}
+		crdName := serverResource.CrdName()
+		_, err := crds.Get(crdName, metav1.GetOptions{})
+		if err == nil {
+			aggregatedStatuses = append(aggregatedStatuses, CrdStatus{Resource: serverResource, ExistenceStatus: Exists})
+			continue
+		}
+		if !errors.IsNotFound(err) {
+			return nil, err
+		}
+		aggregatedStatuses = append(aggregatedStatuses, CrdStatus{Resource: serverResource, ExistenceStatus: NotFound})
+	}
+	return &CrdSummary{Statuses: aggregatedStatuses}, nil
+}
 
-	cmd := &cobra.Command{
-		Use:     "doctor",
-		Aliases: []string{"doc"},
-		Short:   "check riff's requirements are installed",
-		Long: strings.TrimSpace(`
-    <todo>
-    `),
-		Example: "riff doctor",
-		Args:    cli.Args(),
-		PreRunE: cli.ValidateOptions(ctx, opts),
-		RunE:    cli.ExecOptions(ctx, c, opts),
+func checkResourceAccesses(c *cli.Config, checks []AccessChecks) (*AccessSummary, error) {
```

Comment:
Can we merge the existing check with the access check? If the CRD doesn't exist than the access type is "Missing"

Reply:
it's feasible and slightly more efficient, but I find it harder to read.